### PR TITLE
Replace directory with animation_directory in animations

### LINKED ISSUE #5411: Replace `directory` with `animation_directory` in animations
> > Im just saw you used "directory = " in the spritesheets definitions. However our documentation says this is deprecated. https://www.widelands.org/documentation/animations/ So better to use the animation_directory = again.
> 
> I'm sorry, wasn't aware of that. I corrected them for this PR. (but Frisian unit definitions still use `directory =` a lot)
> 
> _Originally posted by @tothxa in https://github.com/widelands/widelands/issues/5376#issuecomment-1122467012_

---

All unit definitions that still use `directory` should be converted to use `animation_directory`.

Support for the old syntax should be kept around for add-ons compatibility, and in order to ensure that it keeps working both syntaxes should be covered in the testsuite.

### PR DESCRIPTION
**Pull request by Veracion**
Fixes #5411 
For v1.2.

- [x] Replace `directory` with `animation_directory` (Exceptions: Frisian soldier, immovables with shared sprites directory)
- [x] Remove `basename` in touched files where it's unnecessary
- [x] Remove outdated TODO comments (see linked issue)
- [x] Coverage in test suite

### CODE REVIEW

**Comment by Veracion** [comment]
For the tests, there is already `test_animations.lua`. It does not cover `directory` yet, however. Should I...
a. move one of the used animations' to a subfolder and add a `directory` to its definition?
b. extend the test by a new unit using only `directory` (without an `animation_directory` entry)?
With the latter one additionally guarding against a potential regression of `animation_directory` actually becoming mandatory. 

For now I've done the first one, which was rather easy to realize.

File: regression_test.py [hunk 1]
```diff
@@ -10,7 +10,6 @@
 import sys
 import tempfile
 import unittest
-import platform
```
Comment (hessenfarmer) [inline regression_test.py:13]: Please explain this deletion

**Comment by hessenfarmer** [comment]
> For the tests, there is already `test_animations.lua`. It does not cover `directory` yet, however. Should I... a. move one of the used animations' to a subfolder and add a `directory` to its definition? b. extend the test by a new unit using only `directory` (without an `animation_directory` entry)? With the latter one additionally guarding against a potential regression of `animation_directory` actually becoming mandatory.
> 
> For now I've done the first one, which was rather easy to realize.

Although I think the solution should work, the safer one would be b) imho (@Noordfrees ). please change accordingly.

**Comment by hessenfarmer** [comment]
Hello @Veracion would you still like to work on this?

File: regression_test.py (hunk 1 above)
Comment (Veracion) [reply regression_test.py:13]: Import was unused

**Comment by Veracion** [comment]
@hessenfarmer 
> Although I think the solution should work, the safer one would be b) imho (@ Noordfrees ). please change accordingly.

Will do, option a didn't work anyways. Test fails because if an `animation_directory` exists for a file, it is always preferred over an individual `directory`.

> would you still like to work on this?

Yes, I'd like to. Will get to it this month.

**Comment by hessenfarmer** [comment]
> > would you still like to work on this?
> 
> Yes, I'd like to. Will get to it this month.

I am glad to hear that

**Comment by hessenfarmer** [comment]
@Veracion any updates on the schedule for this?

**Comment by Veracion** [comment]
Have resolved the merge conflicts and undone the non-working test changes.

Still struggling with the test_animations, progress isn't quite fast. If anyone wants to help or do that part instead I wouldn't mind. Otherwise in the next few weeks, if I get it running.

**Comment by hessenfarmer** [comment]
you might try the most simplistic solution just to duplicate the 2 units in the folder tribes and switch them from animation_directory syntax to directory syntax. just add a "_dir" to their internal name (e.g. barbarians_file_animation_dir) in the init.lua file and the register.lua file.  
duplicate animations_test.lua as well with an appropriate name and use the new names (including the _dir appendix) in the new test file.

**Comment by hessenfarmer** [comment]
Changes look good. if testsuite goes green I'll approve and merge.
Thanks for your contribution.
@Veracion as for every new contributor: how do you wish to be listed in our credits? nickname? realname? or both?

**Comment by Veracion** [comment]
Thank you for the review & patience, hessenfarmer. Glad to have got this done in time for 1.2.

As for the credits, I'd like the nickname only please.

File: test/maps/plain.wmf/scripting/tribes/legacy_file_animation_dir/init.lua [hunk 2]
```diff
@@ -0,0 +1,47 @@
+dirname = "test/maps/plain.wmf/" .. path.dirname(__file__)
+
+wl.Descriptions():new_worker_type {
+   msgctxt = "barbarians_worker",
+   name = "barbarians_legacy_file_animation_dir",
+   -- TRANSLATORS: This is a worker name used in lists of workers
+   descname = pgettext("barbarians_worker", "File Animation"),
+   icon = dirname .. "menu.png",
```
Comment (hessenfarmer) [inline test/maps/plain.wmf/scripting/tribes/legacy_file_animation_dir/init.lua]:
```suggestion
   icon = dirname .. "../file_animation" .. "menu.png",
```

File: test/maps/plain.wmf/scripting/tribes/legacy_spritesheet_animation_dir/init.lua [hunk 3]
```diff
@@ -0,0 +1,82 @@
+dirname = "test/maps/plain.wmf/" .. path.dirname(__file__)
+
+wl.Descriptions():new_productionsite_type {
+   msgctxt = "barbarians_building",
+   name = "barbarians_legacy_spritesheet_animation_dir",
+   -- TRANSLATORS: This is a building name used in lists of buildings
+   descname = pgettext("barbarians_building", "Spritesheet Animation"),
+   icon = dirname .. "menu.png",
```
Comment (hessenfarmer) [inline test/maps/plain.wmf/scripting/tribes/legacy_spritesheet_animation_dir/init.lua]:
```suggestion
   icon = dirname.. "../spritesheet_animation" .. "menu.png",
```

**Review by hessenfarmer — CHANGES_REQUESTED** [review]
found the error for the testsuite I believe

**Comment by hessenfarmer** [comment]
@Veracion please consider commiting my suggestions as I don't want to commit to your fork (not sure whether i am allowed to anyway)

**Comment by Veracion** [comment]
gotcha, am on it

**Review by hessenfarmer — APPROVED** [review]
LGTM now

**Pull request merged**